### DISCUSSION TITLE
Comment implémenter la recherche sur ce projet React ? #1

### DISCUSSION THREAD

**Discussion by jonhsongabriel** [community/community#145642] [Programming Help]
### Body

Bonjours!
Je  suis debutant avec react.
Je cherche à implémenter une fonction de recherche, mais je rencontre des problèmes pour filtrer les résultats dynamiquement. Est-ce que quelqu'un pourrait m'aider à optimiser la logique de filtrage ?
Voila le source de ma projet.
https://github.com/jonhsongabriel/magasin.git

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mecodeatlas** [comment]
Hi @jonhsongabriel,
Thanks for participating!

We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.

**Reply by jonhsongabriel** [reply]
Hello!  
I am a beginner with React.  
I am trying to implement a search function, but I’m having trouble dynamically filtering the results. Could someone help me optimize the filtering logic?  
Here is the source of my project:  
https://github.com/jonhsongabriel/magasin.git

**Comment by jonhsongabriel** [comment]
Hello!  
I am a beginner with React.  
I am trying to implement a search function, but I’m having trouble dynamically filtering the results. Could someone help me optimize the filtering logic?  
Here is the source of my project:  
https://github.com/jonhsongabriel/magasin.git
